variational_autoencoder.py

```diff
@@ -90,10 +85,26 @@ def encode(self, X):
         # Sample from N(mu, sigma)
         self.eps = tf.random_normal(tf.shape(self.Z_mu), dtype=tf.float32, mean=0., stddev=1.0, seed=self.random_seed)
         Z = self.Z_mu + self.Z_sigma * self.eps        
-        # Z = self.Z_mu + self.eps
         return Z
-
-
+    
+    def sample_X(self, age, n):
+        """
+        samples X by first sampling Z from the autoencoder prior, then feeding it through the model. 
+        Draws n samples for people of a given age. 
+        Important note: in our age autoencoder formulation, age is zero-centered 
+        (ie, we train the model with ages whose mean has been subtracted off). 
+        So you probably want the age you pass in to account for that. 
+        """
+        Z = self.sample_Z(age, n)
+        Xr = self.sess.run(self.Xr, feed_dict = {self.Z:Z})
+        # for binary features, need to convert logits to 1s and 0s by sampling. 
+        Xr[:, self.binary_feature_idxs] = np.random.random(Xr[:, self.binary_feature_idxs].shape) < \
+                                                           expit(Xr[:, self.binary_feature_idxs])
+        return Xr
+    
+    def sample_Z(self, age, n):
```

Comment:
Hm, why isn't this implemented? Shouldn't it be sampling from a normal Gaussian?

Reply:
yeah, i just didn't bother to implement this because we were overriding in variational_age_autoencoder anyway. i can add it.